### ISSUE TITLE
How winform designer runs at 100% scale?

### ISSUE THREAD

**Issue by int2e** [dotnet/winforms#2491]
* .NET Core Version: 3.1.100
* Have you experienced this same bug with .NET Framework?: Yes

**Problem description:**
I want to use 100% scale to design UI in the winform designer, my screen resolution is 5120 * 1080 (200% scale);
Now a `Size` property 1000 * 800 is actually 500 * 400;
![size](https://user-images.githubusercontent.com/18641363/70417368-0a678c00-1a9c-11ea-8643-9aa5775c7f34.jpg)

If you let VS start at 100% scale [disable-dpi-awareness](https://docs.microsoft.com/en-us/visualstudio/designers/disable-dpi-awareness?view=vs-2019) will make VS less clear;
I very much hope that the winform designer can run at 100% scale, and F5 runing will not affect DPI scaling;
Just like WPF;

Thanks!

**Comment by RussKie** [comment]
Please provide a repro

**Comment by int2e** [comment]
I provided two videos, winform designer behavior in the range of 100% and 200% respectively
[video.zip](https://github.com/dotnet/winforms/files/3942705/video.zip)

repro
[repro-TestHiDPI.zip](https://github.com/dotnet/winforms/files/3942715/repro-TestHiDPI.zip)

@RussKie

**Comment by RussKie** [comment]
**TL;DR** it is strongly recommended to design your UI at 100% scaling factor (96dpi) and use DPI scaling.


Visual Studio shows a gold bar suggesting to restart with 100% SF
![image](https://user-images.githubusercontent.com/4403806/70501526-63095880-1b72-11ea-8096-3a7f926aeaa4.png)



----

These two lines in your designer are causing your issue:
```cs
this.AutoScaleDimensions = new System.Drawing.SizeF(14F, 31F);
this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
```

If you run your app at 100% SF the true sizes of your controls are:
![image](https://user-images.githubusercontent.com/4403806/70500705-49671180-1b70-11ea-904d-57737fa3bb20.png)

At 200%SF:
![image](https://user-images.githubusercontent.com/4403806/70500287-4b7ca080-1b6f-11ea-865e-b09c5cb0ce25.png)


If you change `this.AutoScaleMode` to `System.Windows.Forms.AutoScaleMode.Dpi` and run the app at 100% SF the form will be sized huge. For me it is 3,860 x 1,677, and the button is 1,371 x 186.
At 200% SF the form is rendered enormous and gets capped at the monitor boundaries (4K for me) - 3,870 x 2,190. And I can't even see the button on the screen, which is sized to 2,743 x 372.


Now if you redesign your form at 100%SF 96dpi but keep the font scaling, you won't get the expected result either:
```cs
this.AutoScaleDimensions = new System.Drawing.SizeF(96F, 96F);
this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
```
![image](https://user-images.githubusercontent.com/4403806/70501108-50425400-1b71-11ea-9cdd-885490538ae1.png)


Changing the `AutoScaleMode` to `AutoScaleMode.Dpi` you will observe the desired results:
![image](https://user-images.githubusercontent.com/4403806/70501070-31dc5880-1b71-11ea-8fb5-6d350fcf19ea.png)


To sum up, everything appears to be working as expected.

---- 



@merriemcgaw we need to verify the new Designer shows the gold bar, as it didn't come up for me for a .NET Core `Form`, but it was shown to me for a .NET Framework `Form`.

**Comment by weltkante** [comment]
Considering that the new designer runs in a separate process I wouldn't expect having to restart VS itself (just the designer process), that may be the reason why the gold bar doesn't come up.

**Comment by RussKie** [comment]
Related #1948

**Comment by merriemcgaw** [comment]
I've added a work item for the Core Designer to ensure the GoldBar shows up.

**Comment by jarenduan** [comment]
@RussKie I met this same problem, but I don't think the goldbar solution is what the developers really want, because the VS UI will become blurry.

I expect the winform designer always runs in 100% scaling factor, while other part of VS always scales with the OS correctly.

Of course, in the manner, the designing size and run time size of the UI element could differ, which seems like not WYSIWYG, however, it eliminates the confusions brought by the "changing size" like this issure mentioned. Winform designer always runs in 100% scaling factor means the size (in pixels) of every UI element won't change with OS scalings, so the developers know the absolute size of UI elements, so they can easily design UI for an expected screen definition.

**Comment by RussKie** [comment]
@merriemcgaw @DustinCampbell

**Comment by merriemcgaw** [comment]
This is a designer/VS feature that is on our radar - not runtime. We plan to do a HDPI push in a later release of .NET and VS. This is what our long term goal is for the designer, but it will take us some time to get there.

**Comment by jarenduan** [comment]
@merriemcgaw  Thanks for the information. It would be the best solution. Looking forward to it!

**Comment by 2mik** [comment]
Currently, is there any lifehack to run WinForm editor without scaling for .NET 6 project on 4K screen?

**Comment by hamarb123** [comment]
Hi, I opened a project today in VS 2022 (17.0.4) and it is certainly not fixed. Even running in 100% scaling mode doesn't fix it. Please remove this from "Done" and fix it, it makes using WinForms almost impossible on High Dpi Displays.

**Comment by 2mik** [comment]
My solution (I hope temporary):
1. Use 2 monitors. The 1st is FullHD, the 2nd is 4K.
2. In display settings, set 1st monitor as primary.
3. Open VS on the 2nd monitor. Despite some troubles, the designer is possible to use.
In my opinion, fixing WinForms designer for high DPI should have a higher priority.

**Comment by hamarb123** [comment]
Thanks @2mik! I've started doing something similar (I run through parallels on a Mac).
I have a 2ndary 1080p monitor and put parallels there in fullscreen and open VS.
I'm yet to use the designer with it moved back, but it would probably work since the designer ignores dpi changes.

**Comment by weltkante** [comment]
> Please remove this from "Done" and fix it

This is the WinForms runtime, you want to report designer issues through VS (the designer github repo isn't public). My guess is that its closed because there is nothing left to do in the WinForms runtime to support this scenario.
